app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java

```diff
@@ -47,6 +67,67 @@ public String getAnswerName() {
         return answerName;
     }
 
+    public void addParent(Question question, String optionUid1) {
+        parents.add(question);
+        Map<String, Boolean> matches = new HashMap<>();
+        if(questionOptionMatches.containsKey(question.getUId())) {
+            matches = questionOptionMatches.get(question.getUId());
+        }
+        matches.put(optionUid1, false);
+        questionOptionMatches.put(question.getUId(), matches);
+    }
+
+    public void addChildren(Question question) {
+        children.add(question);
+    }
+
+    public boolean hasChildren() {
+        return  children.size()>0;
+    }
+    public boolean hasParents() {
+        return  parents.size()>0;
+    }
+
+    public List<Question> getChildren(){
+        return children;
+    }
+
+    public boolean shouldActivateQuestion(QuestionValue value) {
+        Map<String, Boolean> matches = questionOptionMatches.get(value.getQuestionUId());
+        return !isVisible() && matches!=null && matches.containsKey(value.getOptionUId());
+    }
+
+    public void addActiveParentMatch(QuestionValue questionValue) {
+        if(questionValue.getQuestionUId()!=null && questionValue.getOptionUId()!=null){
+            Map<String, Boolean> match = questionOptionMatches.get(questionValue.getQuestionUId());
+            match.put(questionValue.getOptionUId(), true);
+            questionOptionMatches.put(questionValue.getQuestionUId(), match);
+        }
+    }
+
+    public void removeActiveParentMatch(QuestionValue questionValue) {
+        if(questionValue.getQuestionUId()!=null && questionValue.getOptionUId()!=null){
+            Map<String, Boolean> match = questionOptionMatches.get(questionValue.getQuestionUId());
+            match.put(questionValue.getOptionUId(), false);
+            questionOptionMatches.put(questionValue.getQuestionUId(), match);
+        }
+    }
+
+    private boolean hasActiveParent() {
+        for(String questionUId : questionOptionMatches.keySet()){
```

Comment:
I don't understand well this code, please refactor this code to more semantic code, extract in methods if It's necessary where the names are explicative